### ISSUE TITLE
TODO: Speech Recognition Privacy

### ISSUE THREAD

**Issue by snolte26** [snolte26/PyJARVIS#16]
Currently we are using a Google speech recognition API to translate speech into text. Idk how private that API is, so I would like to switch it maybe IBM, or something that works offline

**Comment by brandonland** [comment]
I'd _love_ to help on this one. Some options we have here are:

### Offline solutions
- **Mozilla DeepSpeech**: https://github.com/mozilla/DeepSpeech
  Only downside I can see here is that this uses Google's TensorFlow, but still this is better than hitting Google's API.
- **PocketSphinx**: https://github.com/cmusphinx/pocketsphinx
  There has not been any activity on this project in 3 years, which may be a red flag.
- **Kaldi**: https://github.com/kaldi-asr/kaldi

The downside of going this route is if these engines run on the same machine, I'm not so sure a Raspberry Pi will cut it (maybe the Pi 4). For this reason, it may be best to run the speech engine separately from the machine PyJARVIS runs on. There will need to be some way for PyJARVIS to connect/communicate with the speech server in that case.

The plus side is, the only phoning home that happens is phoning *your* home. :smiley:

**Comment by snolte26** [comment]
Yeah, this one may be a doozy. If only privacy wasn't a main focus
